### ISSUE TITLE
Crash in ChangeTrustOperation and Verification of Sorting Logic

### ISSUE THREAD

**Issue by light-cloud** [Soneso/stellar-ios-mac-sdk#167]
Hi,

Could you please review the following issues?

1. Crash in ChangeTrustOperation class

We are experiencing crashes in the ChangeTrustOperation class. Is it possible to replace the force unwrap here?
Error details:

> stellarsdk/ChangeTrustOperation.swift:46: Fatal error: 'try!' expression unexpectedly raised an error: stellarsdk.StellarSDKError.invalidArgument(message: "Assets are in wrong order. Sort by: Native < AlphaNum4 < AlphaNum12, then by Code, then by Issuer, using lexicographic ordering.")

2. Verification of sorting logic rules

Could you verify the sorting logic rules? It seems that the sorting might need to be case-sensitive.
For example, the following pair causes a crash: USDC/sUSD.
After comparing with the JS Stellar SDK, it appears that sorting should be case-sensitive. In the JS SDK, the `asciiCompare` function is used:
	•	[JS Stellar SDK Asset Comparison](https://github.com/stellar/js-stellar-base/blob/master/src/asset.js#L252)
	•	[JS Stellar SDK Liquidity Pool ID](https://github.com/stellar/js-stellar-base/blob/1036eabf1f20a1154297c419fc2c663040338b22/src/get_liquidity_pool_id.js#L41)

```
/**
 * Compares two ASCII strings in lexographic order with uppercase precedence.
 *
 * @param   {string} a - the first string to compare
 * @param   {string} b - the second
 * @returns {number} like all compare()s:
 *     -1 if a < b, 0 if a == b, and 1 if a > b
 *
 * @warning No type-checks are done on the parameters
 */
function asciiCompare(a, b) {
  return Buffer.compare(Buffer.from(a, 'ascii'), Buffer.from(b, 'ascii'));
}
```

However, in the SDK, it appears the comparison is case-insensitive. Could you confirm this and fix?
Current SDK logic:

```
// Native < AlphaNum4 < AlphaNum12, then by Code, then by Issuer, using lexicographic ordering.
var sortError = false
if assetA.type > assetB.type {
    sortError = true
} else if assetA.type == assetB.type {
    if assetA.code!.caseInsensitiveCompare(assetB.code!) == .orderedDescending {
        sortError = true
    } else if assetA.code!.caseInsensitiveCompare(assetB.code!) == .orderedSame {
        if assetA.issuer!.accountId.caseInsensitiveCompare(assetB.issuer!.accountId) == .orderedDescending {
            sortError = true
        }
    }
}
```

Sample XDR: Remove liquidity pool

> `AAAAAgAAAABn/gGugyO02pFFGaHeLzNEHF4xG6LdYuDPD98S59AjuwADDUABL9thAAAHOQAAAAEAAAAAAAAAAAAAAABnflzGAAAAAAAAAAIAAAAAAAAAF17U/Ee2yWNcLItkCC67OfbsTUVaBlLoBV+SBdMv/ujIAAAAAAAAoHQAAAAAAACgCwAAAAAAAKDeAAAAAAAAAAYAAAADAAAAAAAAAAFVU0RDAAAAADuZETgO/piLoKiQDrHP5E82b32+lGvtB3JA9/Yk3xXFAAAAAXNVU0QAAAAAj2+KyPmYjEMFuwqWpv33O3NZCGoASCtyAGDvCBhfZ/QAAAAeAAAAAAAAAAAAAAAAAAAAAA==`

Thank you for your assistance!

**Comment by christian-rogobete** [comment]
Hi @light-cloud, thank you for adding this issue. I will fix it.

**Comment by christian-rogobete** [comment]
resolved in [release 3.0.3](https://github.com/Soneso/stellar-ios-mac-sdk/releases/tag/3.0.3)
@light-cloud
